internal/service/scheduler/vmscheduler.go

```diff
@@ -45,22 +45,19 @@ func (err InsufficientMemoryError) Error() string {
 
 // ScheduleVM decides which node to a ProxmoxMachine should be scheduled on.
 // It requires the machine's ProxmoxCluster to have at least 1 allowed node.
-func ScheduleVM(ctx context.Context, machineScope *scope.MachineScope) (string, error) {
+func ScheduleVM(ctx context.Context,
+	machineScope *scope.MachineScope,
+	templateMap map[string]int32,
+	allowedNodes []string,
+) (string, int32, error) {
 	client := machineScope.InfraCluster.ProxmoxClient
-	// Use the default allowed nodes from the ProxmoxCluster.
-	allowedNodes := machineScope.InfraCluster.ProxmoxCluster.Spec.AllowedNodes
```

Comment:
why not?

Reply:
this functionality was moved to createVM function to be reused in finding templates